Formula/zsh-syntax-highlighting.rb

```diff
@@ -1,7 +1,7 @@
 class ZshSyntaxHighlighting < Formula
   desc "Fish shell like syntax highlighting for zsh"
   homepage "https://github.com/zsh-users/zsh-syntax-highlighting"
-  url "https://github.com/zsh-users/zsh-syntax-highlighting/archive/0.4.1.tar.gz"
+  url "https://github.com/zsh-users/zsh-syntax-highlighting.git", tag: "0.4.1"
```

Comment:
As an aside, I notice there are a handful of formulae that only specify a tag, with no revision. Would you be interested in a PR that fixed all of these at once by setting the revision to whatever commit the tag is currently pointed to?